src/views/CVERecord/CVERecord.vue

```diff
@@ -78,8 +93,230 @@ export default {
+        }
+      } catch (error) {
+        if (!usecveRecordStore().isPublished) usecveRecordStore().isPublished = false;
+        if (!usecveRecordStore().isReserved) usecveRecordStore().isReserved = false;
+        if (!usecveRecordStore().isRejected) usecveRecordStore().isRejected = false;
+
+        const regex = /4\d{2}/g; // 4xx error, e.g.: 400, 404, etc
+        if (error.toString().match(regex) === null) {
+          this.handleServerError();
+        } else {
+          const errorToStr = error.toString();
+          this.getIdStatusCode = errorToStr.match(regex) ? regex.exec(errorToStr)[0] : 500;
+          if (errorToStr.match(regex)) {
+            usecveRecordStore().isArecord = false;
+            this.handleIsIdOrRecordFound(false);
+          } else {
+            this.handleServerError();
+          }
+        }
+      }
+    },
+    async getRecordData() {
+      const getRecordUrl = `https://${import.meta.env.VITE_CVE_SERVICES_BASE_URL}/api/cve/${usecveRecordStore().cveId}`;
+      const recordData = await axios.get(getRecordUrl);
+
+      if (!this.isJson(recordData.data)) {
+        usecveRecordStore().isPublished = false;
+        usecveRecordStore().isReserved = false;
+        usecveRecordStore().isRejected = false;
+        this.handleServerError();
+      } else if (recordData?.data?.error) {
+        usecveRecordStore().isPublished = false;
+        usecveRecordStore().isReserved = false;
+        usecveRecordStore().isRejected = false;
+
+        if (recordData.data.error === 'CVE_RECORD_DNE') {
+          usecveRecordStore().isIdOrRecordFound = false;
+          usecveRecordStore().isArecord = true;
+        } else {
+          this.handleServerError();
+        }
+      } else {
+        usecveRecordStore().isArecord = true;
+        usecveRecordStore().recordData = recordData.data;
+        if (usecveRecordStore().recordData.cveMetadata.state === 'PUBLISHED') {
+          usecveRecordStore().isPublished = true;
+        } else if (usecveRecordStore().recordData.cveMetadata.state === 'RESERVED') {
+          usecveRecordStore().isReserved = true;
+        } else if (usecveRecordStore().recordData.cveMetadata.state === 'REJECTED') {
+          usecveRecordStore().isRejected = true;
+        }
+        this.handleIsIdOrRecordFound(true);
+      }
+    },
+    handleServerError() {
+      usecveRecordStore().serverError = true;
+      this.setLoadingState(false);
+      usecveRecordStore().isArecord = undefined;
+    },
+    handleIsIdOrRecordFound(newState) {
+      usecveRecordStore().serverError = false;
+      usecveRecordStore().isIdOrRecordFound = newState;
+      this.setLoadingState(false);
+    },
+    setLoadingState(newState) {
+      usecveRecordStore().showHelpText = newState;
+      usecveRecordStore().isSearching = newState;
+      this.disabled = false;
+    },
+    onKeyUpEnter() {
+      this.validateCveId();
+
+      if (!this.disabled) this.startLookup();
+    },
+    isJson(value) {
```

Comment:
This can just be `return typeof value === 'object''` . But since this function is only called in one place, it's probably not even needed. 

Where it's called can be `if (!typeof (recordData.data) === 'object') {`